vllm/config/parallel.py

```diff
@@ -266,19 +270,43 @@ def sync_kv_cache_memory_size(dp_group: ProcessGroup,
 
     def compute_hash(self):
         """
-        Provide a hash that uniquely identifies all the configs
-        that affect the structure of the computation
-        graph from input ids/embeddings to the final hidden states,
-        excluding anything before input ids/embeddings and after
-        the final hidden states.
+        Provide a hash that uniquely identifies parallelism-related configs
+        that affect the compiled graph assumptions.
+
+        Opt-out: default-include declared fields; keep a tiny exclude set;
+        normalize types; keep SHA-256; explicitly include
+        envs.VLLM_ALL2ALL_BACKEND as before.
         """
-        factors: list[Any] = []
-        factors.append(self.pipeline_parallel_size)
-        factors.append(self.tensor_parallel_size)
-        factors.append(self.enable_expert_parallel)
-        factors.append(self.data_parallel_size)
-        factors.append(envs.VLLM_ALL2ALL_BACKEND)
-        return hashlib.sha256(str(factors).encode()).hexdigest()
+
+        EXCLUDE_FROM_HASH = {
+            # Derived/runtime topology, networking, or launch details
+            "data_parallel_rank",
+            "data_parallel_rank_local",
+            "data_parallel_master_ip",
+            "data_parallel_rpc_port",
+            "data_parallel_master_port",
+            "data_parallel_backend",
+            "data_parallel_external_lb",
+            "data_parallel_hybrid_lb",
+            "max_parallel_loading_workers",
+            "disable_custom_all_reduce",
+            "ray_workers_use_nsight",
+            "ray_runtime_env",
+            "placement_group",
+            "distributed_executor_backend",
+            "worker_cls",
+            "sd_worker_cls",
+            "worker_extension_cls",
+            "enable_multimodal_encoder_data_parallel",
```

Comment:
I don't think this should be excluded.  It appears to be used to check whether or not data parallelism should be enabled in a few models.